plugins/opentelemetry/wiring.go

```diff
@@ -1,25 +1,68 @@
+// Once the application is instrumented with these plugins, traces will be generated and collected by the selected collector. Traces will consist of spans generated by Plugin 1 with each span consisting of log events generated by Plugin 2.
+//
+// # Artifacts Generated
+//
+//  1. The package generates client and server side wrappers for instrumented services that contain opentelemetry instrumentation (context propagation, creation of spans). The generated clients handle context propagation correctly on both the server and client sides.
+//
+// # Wiring Usage
+//
+// To instrument services with opentelemetry instrumentation:
+//
+//	opentelemetry.Instrument(spec, "my_service", "collector_name")
+//
+// Calling [Instrument] will generate client and server side wrappers responsible for starting and stopping spans as well as correctly propagating the context across service boundaries.
+//
+// To redirect a process's logging statements to opentelemetry generated spans:
+//
+//	opentelemetry.Logger(spec, "my_process")
+//
+// Calling [Logger] will redirect all logging statements generated by the process to opentelemetry where the log statements will be converted into opentelemetry Events and added to the list of events for the current active span.
+//
+// In order to generate complete end-to-end traces of the application, all services of the application need to be instrumented with OpenTelemetry.
+// If the plugin is only applied to a subset of services, the application will run, but the traces produced won't be end-to-end and won't be useful.
+// The plugin does not support instrumenting clients for backends such as databases, caches, queues, etc. If needed, please consider submitting a PR or contacting the maintainers via google groups or on slack.
+//
+// Example usage (for complete instrumentation):
+//
+// # Wiring Example
+//
+//	func applyOTelOptions() {
+//		jaeger.Container(spec, "jaeger")
+//		for _, service := range serviceNames {
+//			opentelemetry.Instrument(spec, service)
+//		}
+//		for _, proc := range processNames {
+//		opentelemetry.Logger(spec, proc)
+//		}
+//	}
+//
+// See the [ot_logger] wiring spec for the Leaf application for a complete example
+//
+// # Accessing Traces
+//
+// The traces are generated and sent to the zipkin or jaeger collector.
+//
+// [ot_logger]: https://github.com/Blueprint-uServices/blueprint/tree/main/examples/leaf/wiring/specs/custom_logger.go
 package opentelemetry
 
 import (
 	"github.com/blueprint-uservices/blueprint/blueprint/pkg/coreplugins/pointer"
 	"github.com/blueprint-uservices/blueprint/blueprint/pkg/ir"
 	"github.com/blueprint-uservices/blueprint/blueprint/pkg/wiring"
 	"github.com/blueprint-uservices/blueprint/plugins/golang"
+	"github.com/blueprint-uservices/blueprint/plugins/goproc"
 	"golang.org/x/exp/slog"
 )
 
 // Instruments `serviceName` with OpenTelemetry.  This can only be done if `serviceName` is a service declared in the wiring spec using [workflow.Define]
```

Comment:
Add the preamble about being called from a wiring spec.

Reply:
serviceName must be an application-level service, ie. it must not yet be deployed over RPC, or into a process.